Add unittests

Nakaner opened the pull request:
This pull request add some basic unit test for the calculation of the map size, bounding box, scale etc. based on the command line arguments. The unit tests are not complete. Instead, this pull request aims to make the code testable by adding a new class Nik4Image. Otherwise this pull request would become larger than it already is.

This pull request depends on https://github.com/Zverik/Nik4/pull/30 but I am open to rebase it on any other bug fix commit which does the same as https://github.com/Zverik/Nik4/pull/30

See also https://github.com/Zverik/Nik4/issues/31

Nakaner committed:
make calculation of bounding boxes, scales, projections etc. testable

Nakaner committed:
remove doubled assertion

Nakaner committed:
bugfix: function has been moved to the class

Nakaner committed:
add one more unit test

Nakaner committed:
add missing import

Nakaner committed:
add unit tests to check if map size calculation fails

Nakaner committed:
import correct module

Nakaner committed:
add more unit tests for size and scale calculations

Nakaner committed:
make most methods of Nik4Image non-static, decompose setup_options()

Nakaner committed:
split Nik4Image.setup_options into multiple methods

cheshire-mouse commented:
I think, it would be better if Nik4Image class contained everything related to calculations and image building (including run function) and some separate function did argument parsing. This could make possible to use Nik4 as a python module.
It, probably, has nothing to do with unit testing, but since you created a class...

Nakaner committed:
bugfixes

Nakaner committed:
Make code better testable

Nakaner committed:
Improve readbility of tests, add tests

Nakaner committed:
Improve readability

Nakaner committed:
Add more tests

Nakaner committed:
Add tests for projections

Zverik commented:
Thanks for working on this! When you're done, write something, and I'll see if this can be merged :)

There are also some conflicts, check them plz.

Nakaner commented:
I didn't remember that the pull request is still open. :-)

At Geofabrik, I am currently working on a C++ port of Nik4, called Nik5, with the same features and behaviour in order to avoid using Mapnik's Python bindings. That's why I invested time into tests.

Nakaner committed:
Add test for paper sizes

Nakaner committed:
Add test for --fit

Nakaner committed:
Fix initialisation of projections

Mapnik >= 4 expects epsg:1234, < 4 expects +init=epgs:1234.
If Mapnik < 4, it depends on the version of proj. Threfore, the ugly
attempt in the exception handler to try initialisation again.

Nakaner committed:
Fix comparison of projections in order to make tests pass

Nakaner committed:
Add more tests to cover Nik4's behaviour w.r.t. --fit

Nakaner committed:
Add tests for --url

Nakaner committed:
Bugfix: Make parsing of URLs work with positive numbers
